From 16701bed2a07d13e9b4aeaa5ba35bf1d473ec576 Mon Sep 17 00:00:00 2001
From: Miroslav Benes <mbenes@suse.cz>
Date: Mon, 6 Feb 2017 13:40:04 +0100
Subject: [PATCH] kgr: remove all arch-specific kgraft header files
Patch-mainline: not yet, kgraft
References: fate#313296

Function kgr_needs_lazy_migration() is not used anymore (the only caller
was removed with userspace stack detection hack). We omitted to remove its
s390x definition.

arch/s390/include/asm/kgraft.h would be empty without it, so remove it
too. And while doing that, remove empty x86 header file and all the
references too. Everything is handled via livepatch in SLE12-SP2.

Signed-off-by: Miroslav Benes <mbenes@suse.cz>
Reviewed-by: Petr Mladek <pmladek@suse.com>
---
 MAINTAINERS                    |  1 -
 arch/s390/include/asm/kgraft.h | 27 ---------------------------
 arch/x86/include/asm/kgraft.h  |  6 ------
 include/linux/kgraft.h         |  2 --
 4 files changed, 36 deletions(-)
 delete mode 100644 arch/s390/include/asm/kgraft.h
 delete mode 100644 arch/x86/include/asm/kgraft.h

diff --git a/MAINTAINERS b/MAINTAINERS
index cfc0b36b1e45..3fb4507128af 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -6205,7 +6205,6 @@ M:	Jiri Kosina <jkosina@suse.com>
 M:	Jiri Slaby <jslaby@suse.com>
 M:	Vojtech Pavlik <vojtech@suse.com>
 F:	Documentation/kgraft.txt
-F:	arch/*/include/asm/kgraft.h
 F:	include/linux/kgraft.h
 F:	kernel/kgraft*
 F:	samples/kgraft/
diff --git a/arch/s390/include/asm/kgraft.h b/arch/s390/include/asm/kgraft.h
deleted file mode 100644
index fbadece55f73..000000000000
--- a/arch/s390/include/asm/kgraft.h
+++ /dev/null
@@ -1,27 +0,0 @@
-/*
- * kGraft Online Kernel Patching
- *
- *  Copyright (c) 2013-2014 SUSE
- *   Authors: Jiri Kosina
- *	      Vojtech Pavlik
- *	      Jiri Slaby
- */
-
-/*
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License as published by the Free
- * Software Foundation; either version 2 of the License, or (at your option)
- * any later version.
- */
-
-#ifndef ASM_KGR_H
-#define ASM_KGR_H
-
-struct task_struct;
-
-static inline bool kgr_needs_lazy_migration(struct task_struct *p)
-{
-	return true;
-}
-
-#endif
diff --git a/arch/x86/include/asm/kgraft.h b/arch/x86/include/asm/kgraft.h
deleted file mode 100644
index 42d6f93790eb..000000000000
--- a/arch/x86/include/asm/kgraft.h
+++ /dev/null
@@ -1,6 +0,0 @@
-#ifndef ASM_KGR_H
-#define ASM_KGR_H
-
-/* nothing to do here, all handled by LIVEPATCH */
-
-#endif
diff --git a/include/linux/kgraft.h b/include/linux/kgraft.h
index 5baa559807a6..ec40a3f5f58c 100644
--- a/include/linux/kgraft.h
+++ b/include/linux/kgraft.h
@@ -26,8 +26,6 @@
 
 #if IS_ENABLED(CONFIG_KGRAFT)
 
-#include <asm/kgraft.h>
-
 #define KGR_TIMEOUT 2
 
 struct kgr_patch;
-- 
2.12.0

